common/achievement/get.ts

```diff
@@ -0,0 +1,71 @@
+import { prisma } from '../prisma';
+import { Achievement_template, User_achievement, achievement_action_type_enum, achievement_type_enum } from '../../graphql/generated';
+import { Step, Achievement, achievement_state } from '../../graphql/types/achievement';
+
+const getUserAchievements = async (userId: string): Promise<Achievement[]> => {
+    const userAchievements = await prisma.user_achievement.findMany({
+        where: { userId: userId },
+    });
+    const userAchievementGroups: { [group: string]: User_achievement[] } = {};
+    userAchievements.forEach((ua) => {
+        if (!userAchievementGroups[ua.group]) {
+            userAchievementGroups[ua.group] = [];
+        }
+        userAchievementGroups[ua.group].push(ua);
+    });
+    const templates = await prisma.achievement_template.findMany({
+        where: { group: { in: userAchievements.map((ua) => ua.group) } },
```

Comment:
I don't think this is correct, according to https://github.com/corona-school/backend/blob/master/prisma/schema.prisma achievement_template.group is a template whereas user_achievement.group is instantiated. Why not use `user_achievement.templateId` or directly `{ include: { template: true } }` in the user_achievement query?

Reply:
Good idea to include the template in the user_achievement already